### ISSUE TITLE
Migrating to AndroidX

### ISSUE THREAD

**Issue by pavel163** [opacapp/multiline-collapsingtoolbar#62]

**Comment by ssaxena01** [comment]
Does this library have support for AndroidX? 
After migrating to AndroidX, I'm receiving this error:
` Failed to transform artifact 'multiline-collapsingtoolbar.aar (net.opacapp:multiline-collapsingtoolbar:1.6.0)' to match attributes {artifactType=jar}`
Please see related Google Issue: https://issuetracker.google.com/issues/118658603#comment15
This suggests the author provide support for AndroidX.

**Comment by johan12345** [comment]
As discussed in #63 , the update to AndroidX is not a trivial task, as the library is calling a lot of internal support library interfacs which are not part of the public API. Previously, this was not a problem, but with AndroidX, these cannot be called anymore due to the `RestrictTo` annotations.

There are ways to work around this, but that would involve copying a lot of additional classes from AndroidX into our library, which again increases the effort of maintaining it with future updates. Thus, from my point of view, the only viable solution is to finally get Google to implement this in the main Material Components library (see https://github.com/material-components/material-components-android/issues/6).

**Comment by raphaelm** [comment]
I don't have a lot of hope that this will ever happen, but maybe we should create a third feature request here: https://issuetracker.google.com/issues/new?component=439535

**Comment by johan12345** [comment]
We have now opened a PR at https://github.com/material-components/material-components-android/pull/413 and hope this feature can then be merged into the main Material Components library.

**Comment by PattaFeuFeu** [comment]
It seems like we are getting a lot closer‽ The PR has been merged!

I wasn’t able to test the changes just yet but will soon.

**Comment by johan12345** [comment]
Yep, the PR is merged finally :) I'm not completely sure about the next steps, but I suppose it will be included in the next alpha release (`1.2.0-alpha05`?)

**Comment by johan12345** [comment]
Multiline support is included in the [latest alpha release](https://github.com/material-components/material-components-android/releases/tag/1.2.0-alpha05).

However, I found a small bug in this new version that still needs to be fixed:
https://github.com/material-components/material-components-android/issues/1052
